tools/windows/install-help/cal/FinalizeInstall.cpp

```diff
@@ -148,9 +129,17 @@ UINT doFinalizeInstall(CustomActionData &data)
     }
     hr = 0;
 
-    if (!AddUserToRequiredGroups(sid))
-    {
-        goto LExit;
+    if (!data.GetTargetMachine().IsBackupDomainController()) {
```

Comment:
So, this if is not quite correct.
If the customer installs on a BDC first, then they won't get added to the group.

Ideally, what we'd do here is `if(! RODC)`, but I can't find an API that returns that.
What is the error returned from `NetLocalGroupAddMembers()` on a RODC?  I suspect it's generic access denied.  We could add access denied to our list of acceptable errors .

Reply:
Does an RODC report itself as a BDC?

ThenI think as long as we document that when installing on a DC farm, the primary DC has to go first, we're OK.